components/cosmetic_filters/resources/data/element_picker.ts

```diff
@@ -542,14 +556,149 @@ const elementPickerUserModifiedRule = (selector: string) => {
   }
 }
 
+const seShowRulesHiddenBtnState = (
+  showRulesButton: HTMLElement | null, show: boolean) => {
+  if (!showRulesButton) return
+  showRulesButton.textContent = show ? btnHideRulesBoxText : btnShowRulesBoxText
+}
+
+const setMinimizeState = (root: ShadowRoot, minimized: boolean) => {
+  const mainSection = root.getElementById('main-section')
+  if (!mainSection) return;
+  mainSection.classList.toggle('minimized', minimized);
+}
+
+const setupDragging = (root: ShadowRoot): void => {
+  const mainSection = root.getElementById('main-section');
+  const dragHeader = root.getElementById('drag-header');
+
+  if (!mainSection || !dragHeader) return;
+
+  const sectionHeight = mainSection.offsetHeight;
+  const HEADER_HEIGHT = 28;
+
+  let isDragging = false;
+  let startY = 0;
+  let startTransform = 0;
+
+  const parseTranslateY = (transform: string): number => {
+    if (transform === 'none') return 0;
+    const match = transform.match(/matrix\(.*,\s*([-\d.]+)\)$/);
+    if (match) return parseFloat(match[1]);
+    const parts = transform.split(',');
+    return parseFloat(parts[5]) ?? 0;
+  };
+
+  const handleDragStart = (e: TouchEvent): void => {
+    isDragging = true;
+    startY = e.touches[0].clientY;
+    startTransform = parseTranslateY(
+      window.getComputedStyle(mainSection).transform);
+  };
+
+  const handleDragMove = (e: TouchEvent): void => {
+    if (!isDragging) return;
+    e.preventDefault();
+
+    const deltaY = e.touches[0].clientY - startY;
+    const newTransform = Math.min(
+      Math.max(startTransform + deltaY, 0), sectionHeight - HEADER_HEIGHT);
+    mainSection.style.transform = `translateY(${newTransform}px)`;
+  };
+
+  const handleDragEnd = (): void => {
+    if (!isDragging) return;
+    isDragging = false;
+
+    const currentTransform = parseTranslateY(
+      window.getComputedStyle(mainSection).transform);
+    setMinimizeState(root, currentTransform > sectionHeight / 4)
+    mainSection.style.transform = '';
+  };
+
+  dragHeader.addEventListener('touchstart', handleDragStart);
+  root.addEventListener('touchmove', handleDragMove);
+  root.addEventListener('touchend', handleDragEnd);
+};
+
+function initSlider(element: HTMLElement
+  | null, options: SliderOptions = {}): SliderAPI | undefined {
+  if (!element) return;
+
+  const inputElement =  element as HTMLInputElement
+  if (!inputElement) return;
+
+  const min = parseInt(inputElement.min ?? '0', 10);
+  const max = parseInt(inputElement.max ?? '4', 10);
+  const initialValue = 4
+
+  inputElement.tabIndex = 0;
+
+  let currentValue = initialValue;
+
+  const updateSlider = (fireEvent = true): number => {
```

Comment:
probably don't need the default value, as we always specify it
```suggestion
  const updateSlider = (fireEvent: boolean): number => {
```